test/integration/test_as_user.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+from mock import call, patch
+import pytest
+from boxsdk.config import API
+from boxsdk.object.group_membership import GroupMembership
+from boxsdk.object.user import User
+
+
+@pytest.fixture
+def box_api_headers(mock_user_id, access_token):
+    return {'Authorization': 'Bearer {0}'.format(access_token), 'As-User': mock_user_id}
+
+
+def test_client_as_user_causes_as_user_header_to_be_added(
+        box_client,
+        mock_box_network,
+        generic_successful_response,
+        mock_user_id,
+        box_api_headers,
+):
+    # pylint:disable=redefined-outer-name
+    mock_box_network.session.request.side_effect = [
```

Comment:
You can do `.return_value = return_value`, but that's actually different than `.side_effect = [return_value]` because the return_value attribute does additional call tracking that I don't want in this test.